### DISCUSSION TITLE
Unable to use progressfunc to show progress of a download

### DISCUSSION THREAD

**Discussion by demhademha** [curl/curl#6552] [Q&A] [answered]
hi, when trying to use CURLOPT_PROGRESSFUNC, I am unable to get the progress meter. However, the download does succeed.
My code is as follows:
```c
#include <stdio.h>
#include <curl/curl.h>
static int lastprogress = 0;
static int download_progress(void *clientp, double dltotal, double dlnow, double ultotal, double ulnow)
{
	double p = (dlnow / dltotal) * 100;

	if (p < 100.0) {
		if ((int)p > lastprogress) {
                        printf("downloading: %d%%\n", (int)p);
			lastprogress = (int)p;
		}
        }
        return 0;
}


int main(void)
{
  FILE* file = fopen("example.html", "w");

  CURL *curl;
  curl_global_init(CURL_GLOBAL_ALL);
  curl=curl_easy_init();
  curl_easy_setopt(curl, CURLOPT_VERBOSE, 0L);
  curl_easy_setopt(curl, CURLOPT_URL, "https://example.com");
  curl_easy_setopt(curl, CURLOPT_WRITEDATA, file) ;
  curl_easy_setopt(curl, CURLOPT_PROGRESSFUNCTION, (curl_progress_callback)&download_progress);
  curl_easy_perform(curl);

  return 0;
}
```
Any support with this matter would very much be appreciated 
Kind regards

**Comment by bagder** [comment, marked as answer]
It is slightly hidden, but it is mentioned in the [man page](https://curl.se/libcurl/c/CURLOPT_PROGRESSFUNCTION.html):

>  CURLOPT_NOPROGRESS(3)  must  be set to 0 to make this function actually get called.

So add this to your main():
~~~c
  curl_easy_setopt(curl, CURLOPT_NOPROGRESS, 0L);

**Reply by demhademha** [reply]
Unfortunately, disabling the progress metre didn't have any impact.
Do you have any other suggestions? 
Kind regards

**Reply by bagder** [reply]
If you output something unconditionally in your callback you'll probably see that it gets called.

**Reply by demhademha** [reply]
Hi, changing the download url to a larger download did successfully show the download metre. However is it possible , to have the download metre move horizontally rather than vertically? Something like curl -# url. 
Kind regards

**Reply by bagder** [reply]
Your function gets invoked. You decide what it should do!
